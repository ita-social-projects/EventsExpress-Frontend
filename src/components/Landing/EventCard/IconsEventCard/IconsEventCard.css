.heart__icon {
  background: url("../../../../assets/icons/Imgheart.svg") no-repeat center;
}

.join-to-event__icon {
  background: url("../../../../assets/icons/ImgJoinToEvent.svg") no-repeat center;
}

.heart__icon,
.join-to-event__icon {
  height: 40px;
  width: 40px;
  z-index: 1;
  cursor: pointer;
}

.add-favorite-slider {
  top: calc($position-icons-top + $position-top-for-icon-slider);
  left: 30px;
}

.join-to-event-slider {
  top: calc($position-icons-top + $position-top-for-icon-slider);
  left: calc($position-icons-left + $position-left-from-heart-icon);
}

.add-favorite-matrix {
  top: 0;
  left: 0;
}

.join-to-event-matrix {
  top: 205px;
  left: calc($position-icons-left + $position-left-from-heart-icon);
}

.add-favorite-list {
  top: -10px;
  right: 75px;
}

.join-to-event-list {
  top: -10px;
  right: calc($postition-icons-list-right - $position-left-from-heart-icon);
}
/*# sourceMappingURL=IconsEventCard.css.map */