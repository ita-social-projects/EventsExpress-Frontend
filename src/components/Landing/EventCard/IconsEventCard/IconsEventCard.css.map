{
    "version": 3,
    "mappings": "AASA,AAAA,YAAY,CAAC;EACX,UAAU,EAAE,4CAA4C,CAAC,SAAS,CAAC,MAAM;CAC1E;;AAED,AAAA,oBAAoB,CAAC;EACnB,UAAU,EAAE,kDAAkD,CAAC,SAAS,CAAC,MAAM;CAChF;;AAED,AAAA,YAAY;AACZ,oBAAoB,CAAC;EACnB,MAAM,EAAE,IAAI;EACZ,KAAK,EAAE,IAAI;EACX,OAAO,EAAE,CAAC;EACV,MAAM,EAAE,OAAO;CAChB;;AAED,AAAA,oBAAoB,CAAC;EACnB,GAAG,EAAE,yDAAyD;EAC9D,IAAI,EAxBgB,IAAI;CAyBzB;;AAED,AAAA,qBAAqB,CAAC;EACpB,GAAG,EAAE,yDAAyD;EAC9D,IAAI,EAAE,2DAA2D;CAClE;;AAED,AAAA,oBAAoB,CAAC;EACnB,GAAG,EAAE,CAAC;EACN,IAAI,EAAE,CAAC;CACR;;AAED,AAAA,qBAAqB,CAAC;EACpB,GAAG,EAvCgB,KAAK;EAwCxB,IAAI,EAAE,2DAA2D;CAClE;;AAED,AAAA,kBAAkB,CAAC;EACjB,GAAG,EAxCuB,KAAI;EAyC9B,KAAK,EAxCsB,IAAI;CAyChC;;AAED,AAAA,mBAAmB,CAAC;EAClB,GAAG,EA7CuB,KAAI;EA8C9B,KAAK,EAAE,kEAAkE;CAC1E",
    "sources": [
        "IconsEventCard.scss",
        "../../../../styles/colors.scss"
    ],
    "names": [],
    "file": "IconsEventCard.css"
}