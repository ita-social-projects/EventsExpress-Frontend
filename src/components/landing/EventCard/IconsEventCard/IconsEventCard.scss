@import "../../../../styles/colors.scss";

$position-icons-top: 205px;
$position-icons-left: 30px;
$position-left-from-heart-icon: 50px;
$position-top-for-icon-slider: 0px;
$postition-icons-list-top: -10px;
$postition-icons-list-right: 75px;

.heart__icon {
  background: url("../../../../assets/icons/Imgheart.svg") no-repeat center;
}

.join-to-event__icon {
  background: url("../../../../assets/icons/ImgJoinToEvent.svg") no-repeat
    center;
}

.heart__icon,
.join-to-event__icon {
  position: absolute;
  height: 40px;
  width: 40px;
  background-color: $lightThemeAccept;
  box-shadow: 0px 4px 4px $transparentBlackBackground;
  z-index: 1;
  cursor: pointer;
}

.add-favorite-slider {
  top: calc($position-icons-top + $position-top-for-icon-slider);
  left: $position-icons-left;
}

.join-to-event-slider {
  top: calc($position-icons-top + $position-top-for-icon-slider);
  left: calc($position-icons-left + $position-left-from-heart-icon);
}

.add-favorite-matrix {
  top: $position-icons-top;
  left: $position-icons-left;
}

.join-to-event-matrix {
  top: $position-icons-top;
  left: calc($position-icons-left + $position-left-from-heart-icon);
}

.add-favorite-list {
  top: $postition-icons-list-top;
  right: $postition-icons-list-right;
}

.join-to-event-list {
  top: $postition-icons-list-top;
  right: calc($postition-icons-list-right - $position-left-from-heart-icon);
}
