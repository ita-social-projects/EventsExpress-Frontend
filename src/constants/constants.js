// this constant is used to get a token from localStorage
export const jwtStorageKey = "token";

export default jwtStorageKey;
