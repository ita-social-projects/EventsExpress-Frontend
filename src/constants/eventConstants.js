export const EVENT_DEFAULT_IMAGE =
  "data:image/svg+xml,%3Csvg%20width%3D%22344%22%20height%3D%22194%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20xmlns%3Axlink%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%22%3E%3Cdefs%3E%3Cpath%20id%3D%22a%22%20d%3D%22M-1%200h344v194H-1z%22%2F%3E%3C%2Fdefs%3E%3Cg%20transform%3D%22translate(1)%22%20fill%3D%22none%22%20fill-rule%3D%22evenodd%22%3E%3Cmask%20id%3D%22b%22%20fill%3D%22%23fff%22%3E%3Cuse%20xlink%3Ahref%3D%22%23a%22%2F%3E%3C%2Fmask%3E%3Cuse%20fill%3D%22%23BDBDBD%22%20xlink%3Ahref%3D%22%23a%22%2F%3E%3Cg%20mask%3D%22url(%23b)%22%3E%3Cpath%20d%3D%22M173.65%2069.238L198.138%2027%20248%20112.878h-49.3c.008.348.011.697.011%201.046%200%2028.915-23.44%2052.356-52.355%2052.356C117.44%20166.28%2094%20142.84%2094%20113.924c0-28.915%2023.44-52.355%2052.356-52.355%2010%200%2019.347%202.804%2027.294%207.669zm0%200l-25.3%2043.64h50.35c-.361-18.478-10.296-34.61-25.05-43.64z%22%20fill%3D%22%23757575%22%2F%3E%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E";

export const LEAVE_MODAL = {
  LEAVE: "Leave",
  EXIT_EVENT: "Exiting from event",
  ALERT_LEAVE_EVENT:
    'Are you sure that you want to leave this event. If you leave, your statement will be deleted. Press "Agree" if you want to leave  event and "Disagree" if not.`',
  AGREE: "Agree",
  DISAGREE: "Disagree",
};

export const ENUM_LOCATION_TYPE = {
  MAP: 0,
  ONLINE: 1,
};

export const FORMATS = {
  MONTH_FORMAT: "MMMM",
  DAY_FORMAT: "DD",
};

export const EVENTS_FILTER = {
  RADIUS: "Radius:",
  RADIUS_KM: "km",
  LOCATION: "Location:",
  BUTTON_LESS: "less...",
  BUTTON_MORE_FILTERS: "more filters...",
  BUTTON_RESET: "Reset",
  BUTTON_FAVORITE: "Favorite",
  BUTTON_SEARCH: "Search",
  OPTIONS_ACTIVE: "Active",
  OPTIONS_BLOCKED: "Blocked",
  OPTIONS_CANCELED: "Canceled",
};

export const EVENT_ITEM_CONSTANTS = {
  PARTICIPANTS: "Participants",
};

export const EVENT_STATUS_BUTTON = {
  TEXT_SUCCESS: "text-success",
  TEXT_DANGER: "text-danger",
};

export const EVENT_STATUS_ENUM = {
  ACTIVE: 0,
  BLOCKED: 1,
  CANCELED: 2,
  DRAFT: 3,
  DELETED: 4,
};

export const EVENT_STATUS_ICON = {
  ACTIVE: "fas fa-unlock",
  BLOCKED: "fas fa-lock",
  CANCELED: "far fa-calendar-times",
};

export const EVENT_STATUS_TITLE = {
  ACTIVE: "Active event",
  BLOCKED: "Blocked event",
  CANCELED: "Canceled event",
};

export const EVENT_VIEW_MODE_MAPPER = [
  { id: 1, control: `menu-view-mode__matrix`, viewMode: "matrix" },
  { id: 2, control: `menu-view-mode__list`, viewMode: "list" },
  { id: 3, control: `menu-view-mode__slider`, viewMode: "slider" },
];

export const VIEW_MODE_TYPES = {
  MATRIX: "matrix",
  LIST: "list",
  SLIDER: "slider",
};

export const VIEW_MODE_KEY_FOR_LOCAL_STORAGE = "viewModeEvents";

export const EVENT_VISITORS = {
  ADMIN: "Admin",
  VISITORS: "Visitors",
  PENDING_USERS: "Pending users",
  DENIED_USERS: "Denied users",
};

export const CARD_TYPE = {
  DRAFT: "draft",
  LANDING: "landing",
  HOME: "home",
  RECCURENT_EVENTS: "eventSchedules",
};
export const EVENT_NO_RESULT = {
  TITLE: "–ùou have no events yet",
  SUB_TITLE: "You have not yet created events with the ability to repeat them",
  PHOTO:
    "https://res.cloudinary.com/wunu/image/upload/v1654537602/eventexpress/folder-is-empty-4064360-3363921_y9cgvg.png",
  BTN_TITLE: "Back",
};

export const CANCEL_EVENTS = {
  CANCEL_ONE: "Cancel one",
  CANCEL: "Cancel",
};

export const CREATE_WITHOUT_EDITING = "Create without editing";
export const CREATE_WITH_EDITING = "Create with editing";
export const CONFIRMATION = "Confirmation";
export const ENTER_LOCATION = "Enter an https:// URL:";
export const CURRENT_POSITION_ON_MAP = "Current position on the Map is:";
export const ONLINE_MEETING_HERE = "Online meeting here";
export const RUN = {
  LAST: "Last run",
  NEXT: "Next run",
};
export const FILTERS = "Filters";

export const EVENT_ITEM_VIEW_CONSTS = {
  YOU_GONNA_VISIT: "You are gonna visit.",
  DENIED_PARTICIPATION: "Denied participation.",
  WAIT_ADMIN_APROVE: "Wait until admin approve your request.",
  YOU_NOT_IN_EVENT: "You are not in event yet.",
  PARTICIPANTS: "Participants",
  PUBLIC_EVENT: "Public event",
  PRIVATE_EVENT: "Private event",
  ADULT_LABEL: "18+",
  EVENTS_FOR_ADULTS: "This event is only for adults.",
  AGE_REQUIREMENTS: "You do not meet age requirements for this event.",
};

export const EVENT_SCHEDULE_CONSTS = {
  CLICK: "Click",
  CANCEL_ONE: "Cancel Once",
  CANCEL: "Cancel",
  EDITING: "editing.",
  TO_CREATE_WITH: "to create the event with",
  TO_CREATE_WITHOUT: "to create the event without",
  TO_CANCEL_ALL: "to cancel all events.",
  TO_CANCEL_NEXT: "to cancel the next event.",
};

export const EVENT_CARD_TITLE_SLICE = 30;
export const EVENT_LIST_TITLE_SLICE = 35;
export const EVENT_LIST_DESC_SLICE = 100;

export const EVENT_ITEM_DESC_SLICE = 570;
export const EVENT_ITEM_MAX_DESC = 128;

export const EVENT_OPACITY = {
  HALF: 0.5,
  FULL: 1,
};

export const EVENT_FILTER_Z_INDEX = 2;

export const EVENT_SLICE_CATEGORIES_PARAM = 2;
