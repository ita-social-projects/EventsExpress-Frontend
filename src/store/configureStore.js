import { applyMiddleware, combineReducers, compose, createStore } from "redux";
import thunk from "redux-thunk";
import { routerMiddleware } from "react-router-redux";
import rootReducers from "../reducers/root";

export default function configureStore(history, initialState) {
  const middleware = [thunk, routerMiddleware(history)];

  // In development, use the browser's Redux dev tools extension if installed
  const isDevelopment = process.env.NODE_ENV === "development";
  const composeEnhancers =
    isDevelopment &&
    typeof window !== "undefined" &&
    (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose);
  const rootReducer = combineReducers({
    ...rootReducers,
  });

  return createStore(
    rootReducer,
    initialState,
    composeEnhancers(applyMiddleware(...middleware)),
  );
}
