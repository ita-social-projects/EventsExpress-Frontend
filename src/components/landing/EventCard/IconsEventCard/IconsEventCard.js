import React from "react";
import "./IconsEventCard.scss";

const IconsEventCard = ({ styleForIcon }) => {
  const MAPPER = [
    {
      id: 1,
      nameControl: "favorite",
      styles: `heart__icon add-favorite-${styleForIcon}`,
      todo: "TODO: CREATE FUNCTIONAL FOR ADD FAVORITE EVENT",
    },
    {
      id: 2,
      nameControl: "joinToEvent",
      styles: `join-to-event__icon join-to-event-${styleForIcon}`,
      todo: "TODO: CREATE FUNCTIONAL FOR JOIN TO EVENT",
    },
  ];

  return MAPPER.map(({ id, nameControl, styles, todo }) => (
    <label key={id} htmlFor={nameControl} className={styles}>
      <button
        type="button"
        name={nameControl}
        id={nameControl}
        onClick={() => {
          console.log(todo);
        }}
      >
        {" "}
      </button>
    </label>
  ));
};

export default IconsEventCard;
