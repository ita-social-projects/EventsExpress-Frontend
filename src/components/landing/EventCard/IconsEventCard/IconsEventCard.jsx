import React from "react";
import { connect } from "react-redux";
import "./IconsEventCard.scss";

const IconsEventCard = ({ styleForIcon, userId }) => {
  const MAPPER = [
    {
      id: 1,
      visibility: true,
      nameControl: "favorite",
      styles: `heart__icon add-favorite-${styleForIcon}`,
      todo: "TODO: CREATE FUNCTIONAL FOR ADD FAVORITE EVENT",
    },
    {
      id: 2,
      visibility: userId,
      nameControl: "joinToEvent",
      styles: `join-to-event__icon join-to-event-${styleForIcon}`,
      todo: "TODO: CREATE FUNCTIONAL FOR JOIN TO EVENT",
    },
  ];

  return MAPPER.map(
    ({ id, nameControl, styles, visibility }) =>
      visibility !== null && (
        <label key={id} htmlFor={nameControl} className={styles}>
          <button
            type="button"
            name={nameControl}
            id={nameControl}
            onClick={() => {
              // console.log(todo);
            }}
          >
            {" "}
          </button>
        </label>
      ),
  );
};

const mapStateToProps = ({ user }) => ({ userId: user.id });

export default connect(mapStateToProps, {})(IconsEventCard);
