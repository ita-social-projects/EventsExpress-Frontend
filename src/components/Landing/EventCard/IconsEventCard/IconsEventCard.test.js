import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { VIEW_MODE_TYPES } from "../../../../constants/eventConstants";
import IconsEventCard from "./IconsEventCard";

const { MATRIX, SLIDER, LIST } = VIEW_MODE_TYPES;

const mockStore = configureStore([]);

const treeWithViewMode = (store, viewMode) =>
  renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <IconsEventCard styleForIcon={viewMode} />
      </Provider>
      ,
    </BrowserRouter>,
  );

describe("renders icons in different view modes when user logged in", () => {
  const store = mockStore({
    user: {
      id: true,
    },
  });

  it("should return matrix view mode with two icons", () => {
    const tree = treeWithViewMode(store, MATRIX).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should return list view mode with two icons", () => {
    const tree = treeWithViewMode(store, LIST).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should return slider view mode with two icons", () => {
    const tree = treeWithViewMode(store, SLIDER).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("renders icons in different view modes when user is not logged in", () => {
  const store = mockStore({
    user: {
      id: null,
    },
  });

  it("should return matrix view mode with one icon", () => {
    const tree = treeWithViewMode(store, MATRIX).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should return list view mode with one icon", () => {
    const tree = treeWithViewMode(store, LIST).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should return slider view mode with one icon", () => {
    const tree = treeWithViewMode(store, SLIDER).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
