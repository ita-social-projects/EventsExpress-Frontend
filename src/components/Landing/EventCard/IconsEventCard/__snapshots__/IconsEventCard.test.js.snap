// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders icons in different view modes when user is not logged in should return list view mode with one icon 1`] = `
Array [
  <label
    className="heart__icon add-favorite-list"
    htmlFor="favorite"
  >
    <button
      id="favorite"
      name="favorite"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </label>,
  ",",
]
`;

exports[`renders icons in different view modes when user is not logged in should return matrix view mode with one icon 1`] = `
Array [
  <label
    className="heart__icon add-favorite-matrix"
    htmlFor="favorite"
  >
    <button
      id="favorite"
      name="favorite"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </label>,
  ",",
]
`;

exports[`renders icons in different view modes when user is not logged in should return slider view mode with one icon 1`] = `
Array [
  <label
    className="heart__icon add-favorite-slider"
    htmlFor="favorite"
  >
    <button
      id="favorite"
      name="favorite"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </label>,
  ",",
]
`;

exports[`renders icons in different view modes when user logged in should return list view mode with two icons 1`] = `
Array [
  <label
    className="heart__icon add-favorite-list"
    htmlFor="favorite"
  >
    <button
      id="favorite"
      name="favorite"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </label>,
  <label
    className="join-to-event__icon join-to-event-list"
    htmlFor="joinToEvent"
  >
    <button
      id="joinToEvent"
      name="joinToEvent"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </label>,
  ",",
]
`;

exports[`renders icons in different view modes when user logged in should return matrix view mode with two icons 1`] = `
Array [
  <label
    className="heart__icon add-favorite-matrix"
    htmlFor="favorite"
  >
    <button
      id="favorite"
      name="favorite"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </label>,
  <label
    className="join-to-event__icon join-to-event-matrix"
    htmlFor="joinToEvent"
  >
    <button
      id="joinToEvent"
      name="joinToEvent"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </label>,
  ",",
]
`;

exports[`renders icons in different view modes when user logged in should return slider view mode with two icons 1`] = `
Array [
  <label
    className="heart__icon add-favorite-slider"
    htmlFor="favorite"
  >
    <button
      id="favorite"
      name="favorite"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </label>,
  <label
    className="join-to-event__icon join-to-event-slider"
    htmlFor="joinToEvent"
  >
    <button
      id="joinToEvent"
      name="joinToEvent"
      onClick={[Function]}
      type="button"
    >
       
    </button>
  </label>,
  ",",
]
`;
